daemon/core.cpp

```diff
@@ -1399,6 +1399,7 @@ void Core::run()
                                     log(LOG_CRIT, "Cannot read from X11 request pipe: %s", strerror(error));
                                     close(mX11ResponsePipe[STDIN_FILENO]);
                                     mX11EventLoopActive = false;
+                                    delete[] str;
```

Comment:
How about removing all such "old fashion" new/delete and using smart pointers (QT or std...whatever)?